client/web/src/components/series/label/SeriesLabel.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { useDispatch } from 'react-redux'
+import {
+  Chip,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  Box,
+} from '@material-ui/core'
+import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
+
+import { actionTypes, SeriesService } from '../../../features/series'
+import { ILabel } from '../../../models/ILabel'
+import { ISeries } from '../../../models/series'
+import { SeriesLabelFrom } from './SeriesLabelFrom'
+
+type SeriesLabelProps = { series: ISeries }
+
+export const SeriesLabel: React.FC<SeriesLabelProps> = (props: SeriesLabelProps) => {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  let labels = props.series.labels;
+
+  // For form
+  const [open, setOpen] = React.useState(false);
+  const handleClickOpen = (localSeries: ISeries | null) => {
+    setOpen(true);
+    dispatch({ type: actionTypes.SET_CURRENT_SERIES, series: localSeries })
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handleDelete = (localSeries: ISeries | null, labelClicked: ILabel) => {
+    if(localSeries == null)
+    {
+      throw Error("No series selected")
+    }
+    if(localSeries.labels == null)
+    {
+      localSeries.labels = new Array<ILabel>();
+    }
+    localSeries.labels.forEach((label: ILabel) => {
+      var newLabels = localSeries.labels.filter((label) => { 
+        if(label !== labelClicked){
+          return label;
+        }
+        return null;
+      });
+      localSeries.labels = newLabels;
+    });
+    SeriesService.update(localSeries.id, localSeries).then(()=> {dispatch({ type: actionTypes.FETCH_ALL_SERIES })});
```

Comment:
You can use redux-saga here :

In seriesSaga.ts : 
```typescript
function* updateSeries(action) {
  const { series } = action
  yield call([SeriesService, 'update'], series.id, series)
  yield call(fetchSeries)
}
// You have to update actionWatcher function with
yield takeLatest(actionTypes.UPDATE_SERIES, updateSeries) //Add actionType UPDATE_SERIES
}
```
In feature/user/types.ts, you must add :
```typescript
interface UpdateSeriesAction {
  type: typeof UPDATE_SERIES
  series: ISeries
}

// Add to type SeriesActionTypes
```

finally, you just can call in that file the line below:
```typescript
dispatch({ type: UPDATE_SERIES, series: localSeries
```

Reply:
Done